cirq/ops/common_gates.py

```diff
@@ -515,11 +551,19 @@ def matrix(self):
         s = math.sqrt(0.5)
         return np.array([[s, s], [s, -s]])
 
+    def text_diagram_info(self, args: gate_features.TextDiagramInfoArgs
+                          ) -> gate_features.TextDiagramInfo:
+        return gate_features.TextDiagramInfo(('H',))
+
     def known_qasm_output(self,
                           qubits: Tuple[raw_types.QubitId, ...],
                           args: gate_features.QasmOutputArgs) -> Optional[str]:
         args.validate_version('2.0')
-        return args.format('h {0};\n', qubits[0])
+        if self.half_turns == 1:
+            return args.format('h {0};\n', qubits[0])
+        else:
+            return args.format('rh({0:half_turns}) {1};\n',
```

Comment:
I know the other cases don't do this yet, but could you add a comment linking to external documentation stating that `rh` exists and what it is?

Reply:
Should we change the name of this class from `HGate` to  `RotHGate`? It will break a couple of things, but that is what it does now.

Reply:
I like @dabacon 's idea to rename to things like `HPowGate`

Reply:
`XPowGate`, `YPowGate` and `ZPowGate` too?

Reply:
Yes, but don't do that in this PR.

Reply:
>Add a comment linking to external documentation stating that rh exists and what it is?

I haven't been able to find external documentation about `rh`. I;m not sure where to add it. Maybe in the `Single Qubit Gates` section? https://github.com/quantumlib/Cirq/blob/master/docs/api.rst

`rh` is qasm. Does a user need to know about it for something like `cirq.H(q0) ** 0.25` which compiles to `rh`? Maybe emulating how the Rotation gates are documented will suffice. 

For example:
`HPowGate(*, half_turns, float, …) | Fixed rotation around the X+Z axis of the Bloch sphere.`

Reply:
How did you find out about `rh` in the first place if you never found any documentation on it?

Reply:
It didn't exist before.  This pull request is `rh`'s implementation hence the

```
if self.half_turns == 1:
            return args.format('h {0};\n', qubits[0])
        else:
            return args.format('rh({0:half_turns}) {1};\n',
                               self.half_turns, qubits[0])
```

Reply:
What do you mean that "this is rh's implementation"? Is rh not an existing function in QASM? The QASM output code is supposed to produce valid QASM that can run anywhere; we can't just make up our own functions.

Reply:
Update: Hardcoded the generated code instead of calling `default_compose()` because that would require calling `QasmOutput()`  which can't called recursively as it generates the same header and qreg twice. This is what I was planning on doing

```
 def known_qasm_output(self,
                          qubits: Tuple[raw_types.QubitId, ...],
                          args: gate_features.QasmOutputArgs) -> Optional[str]:
        args.validate_version('2.0')

        lst = list(self.default_decompose(qubits))
        output = qasm_output.QasmOutput(lst, tuple(qubits))
        
        return str(output)
```